crates/forge/tests/cli/cmd.rs

```diff
@@ -2390,6 +2390,136 @@ contract CounterTest is DSTest {
     );
 });
 
+// <https://github.com/foundry-rs/foundry/issues/9115>
+forgetest_init!(gas_report_with_fallback, |prj, cmd| {
+    prj.add_test(
+        "DelegateProxyTest.sol",
+        r#"
+import {Test} from "forge-std/Test.sol";
+
+contract ProxiedContract {
+    uint256 public amount;
+
+    function deposit(uint256 aba) external {
+        amount = amount * 2;
+    }
+
+    function deposit() external {
+    }
```

Comment:
It the shadowing of `deposit` here intended?

Reply:
yeah, to make sure that if deposit fn with different params exists on proxy contract then gas is accounted for proxy (that is we match on function signatures and not on function names).